cartography/intel/aws/kms.py

```diff
@@ -0,0 +1,333 @@
+    except ClientError as e:
+        policy = None
+        logger.warning("Failed to retrieve Key Policy for key id - {}. Error - {}".format(key["KeyId"], e))
+        raise
+
+    return policy
+
+
+@timeit
+def get_aliases(key, client):
+    """
+    Gets the KMS Key Aliases.
+    """
+    aliases = []
+    paginator = client.get_paginator('list_aliases')
+    for page in paginator.paginate(KeyId=key['KeyId']):
+        aliases.extend(page['Aliases'])
+
+    return aliases
+
+
+@timeit
+def get_grants(key, client):
+    """
+    Gets the KMS Key Grants.
+    """
+    grants = []
+    paginator = client.get_paginator('list_grants')
+    for page in paginator.paginate(KeyId=key['KeyId']):
+        grants.extend(page['Grants'])
+
+    return grants
+
+
+@timeit
+def _load_kms_key_aliases(neo4j_session, aliases, update_tag):
+    """
+    Ingest KMS Aliases into neo4j.
+    """
+    ingest_aliases = """
+    MERGE (a:KMSAlias{id: {AliasArn}})
+    ON CREATE SET a.firstseen = timestamp(), a.targetkeyid = {TargetKeyId}
+    SET a.aliasname = {AliasName}, a.lastupdated = {UpdateTag}
+    WITH a MATCH (kmskey:KMSKey{id: {TargetKeyId}})
+    MERGE (a)-[r:KNOWN_AS]->(kmskey)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {UpdateTag}
+    """
+
+    for alias in aliases:
+        neo4j_session.run(
+            ingest_aliases,
+            AliasArn=alias["AliasArn"],
+            AliasName=alias.get("AliasName"),
+            TargetKeyId=alias["TargetKeyId"],
+            UpdateTag=update_tag,
+        )
+
+
+@timeit
+def _load_kms_key_grants(neo4j_session, grants_list, update_tag):
+    """
+    Ingest KMS Key Grants into neo4j.
+    """
+    ingest_grants = """
+    MERGE (g:KMSGrant{id: {GrantId}})
+    ON CREATE SET g.firstseen = timestamp(), g.granteeprincipal = {GranteePrincipal}, g.creationdate = {CreationDate}
+    SET g.name = {GrantName}, g.lastupdated = {UpdateTag}
+    WITH g
+    MATCH (kmskey:KMSKey{id: {TargetKeyId}})
+    MERGE (g)-[r:APPLIED_ON]->(kmskey)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {UpdateTag}
+    """
+
+    for grant in grants_list:
+        neo4j_session.run(
+            ingest_grants,
+            GrantId=grant["GrantId"],
+            GrantName=grant.get("Name"),
+            GranteePrincipal=grant.get("GranteePrincipal"),
+            CreationDate=grant["CreationDate"],
+            TargetKeyId=grant["KeyId"],
+            UpdateTag=update_tag,
+        )
+
+
+@timeit
+def _load_kms_key_policies(neo4j_session, policies, update_tag):
+    """
+    Ingest KMS Key policy results into neo4j.
+    """
+    # NOTE we use the coalesce function so appending works when the value is null initially
+    ingest_policies = """
+    UNWIND {policies} AS policy
+    MATCH (k:KMSKey) where k.name = policy.kms_key
+    SET k.anonymous_access = (coalesce(k.anonymous_access, false) OR policy.internet_accessible),
```

Comment:
Your schema documentation should include what `kmskey.anonymous_access` means.

Reply:
schema updated